manifests/allow_client_on_server.pp

```diff
@@ -0,0 +1,56 @@
+# @summary Enable backup client to access backup server
+#
+# @param hostname
+#   fqdn hostname of backup client.
+#
+# @param ip
+#   ip of backup client.
+#
+# @param key
+#   The public ssh key itself; generally a long string...
+#
+# @param key_type
+#   The encryption type used in the public ssh key.
+#
+# @example
+#   profile_backup::allow_client_on_server { 'allow host hostname access to backup servers':
+#     'hostname' => String,
+#     'ip' => String,
+#     'key' => String,
+#     'key_type' => String,
+#   }
+define profile_backup::allow_client_on_server (
+  String $hostname,
+  String $ip,
+  String $key,
+  String $key_type,
+) {
+  # SET DEFAULTS FOR CLIENT
+  $repo_path = "${profile_backup::server::backup_directory}/${hostname}"
+
+  # SETUP SSHD MATCHBLOCK, PAM ACCESS, & FIREWALL FOR CLIENT
+  ::sshd::allow_from { $name:
+    additional_match_params => $profile_backup::server::additional_sshd_match_params,
+    hostlist                => [$ip],
+    users                   => [$profile_backup::server::username],
+  }
+
+  # SETUP SSH AUTHORIZED KEY ENTRY FOR CLIENT
+  ssh_authorized_key { $hostname:
+    ensure  => present,
+    key     => $key,
+    # SEE https://borgbackup.readthedocs.io/en/stable/quickstart.html#remote-repositories
+    options => [
+      "from=\"${ip}\"",
+      "command=\"borg serve --restrict-to-path ${repo_path}\"",
+      'restrict',
+    ],
+    type    => $key_type,
+    user    => $profile_backup::server::username,
+  }
+
+  # REPOSITORY DIRECTORY IN TAIGA
```

Comment:
remove TAIGA from the comment; this profile is storage agnostic, yes?

Reply:
done